read.go

```diff
@@ -573,7 +573,7 @@ func NewSearcher(r IndexFile) (Searcher, error) {
 		return nil, err
 	}
 	indexData.file = r
-	return indexData, nil
+	return &tenantAwareSearcher{d: indexData}, nil
```

Comment:
a simplification: rather than making tenantAwareSearcher check EnforcementMode on every request, only wrap if enforcement mode is on. I think that better justifies creating this wrapper. Otherwise I'd prefer removing this wrapper and moving the checks into indexData.